fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -376,20 +379,25 @@ private CompletableFuture<RecordCursorResult<IndexEntry>> switchToNextPartitionA
         if (sortedByPartitioningKey && !isReverseSort) {
             // if we're in a partitioning-field-ascending-sort query, get the next more recent partition
             nextPartitionFuture = partitioner.getPartitionMetaInfoById(partitionId, groupingKey)
-                    .thenCompose(curPartition -> LucenePartitioner.getNextNewerPartitionInfo(state.context, groupingKey, partitionTimestamp, state.indexSubspace));
+                    .thenCompose(curPartition -> LucenePartitioner.getNextNewerPartitionInfo(state.context, groupingKey, partitioner.getPartitionKey(curPartition), state.indexSubspace));
         } else {
             // otherwise get the next older partition
-            nextPartitionFuture = partitioner.findPartitionInfo(Objects.requireNonNull(groupingKey), partitionTimestamp - 1);
+            nextPartitionFuture = LucenePartitioner.getNextOlderPartitionInfo(
+                    state.context,
+                    Objects.requireNonNull(groupingKey),
+                    partitionKey,
+                    state.indexSubspace);
         }
 
         return nextPartitionFuture.thenCompose(nextPartition -> {
             if (nextPartition != null && nextPartition.getId() != partitionId) {
                 // reset scan params/state
                 exhausted = false;
-                this.partitionTimestamp = LucenePartitioner.getFrom(nextPartition);
+                this.partitionKey = partitioner.getPartitionKey(nextPartition);
                 this.partitionId = nextPartition.getId();
                 if (!dontResetSearchAfter) {
                     searchAfter = null;
+                    searchAfterPartitioningKey = null;
```

Comment:
I think this is correct, was it just a bug before, or is something semantic changing in the surrounding code that requires this clearing to be explicit here?

Reply:
I think the previous version should have also reset `searchAfterTimestamp`